src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -0,0 +1,396 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Intrinsics.X86;
+using System;
+
+namespace Microsoft.ML.Runtime.Internal.CpuMath
+{
+    public static partial class CpuMathUtils
+    {
+        public static void Scale(float a, float[] dst, int count)
+        {
+            Contracts.AssertNonEmpty(dst);
+            Contracts.Assert(0 < count && count <= dst.Length);
+
+            Scale(a, new Span<float>(dst, 0, count));
+        }
+
+        public static void Scale(float a, float[] dst, int offset, int count)
```

Comment:
Also for future cleanup: It's probably worth just exposing a public `Span<T>` overload, and letting the external code set offset/count as needed.

Reply:
That would involve changing the other `partial` class as well, which is used for `netstandard2.0`. It would also mean modifying all the call sites. For sure, we can do that in the future.